pygmt/helpers/tempfile.py

```diff
@@ -126,18 +126,27 @@ def tempfile_from_geojson(geojson):
         E.g. '1a2b3c4d5e6.gmt'.
     """
     with GMTTempFile(suffix=".gmt") as tmpfile:
+        # pylint: disable=import-outside-toplevel
+        import geopandas as gpd
+
         os.remove(tmpfile.name)  # ensure file is deleted first
         ogrgmt_kwargs = {"filename": tmpfile.name, "driver": "OGR_GMT", "mode": "w"}
         try:
+            # Workaround to map int/int64 to int32
+            # https://github.com/geopandas/geopandas/issues/967#issuecomment-842877704
+            # https://github.com/GenericMappingTools/pygmt/issues/2497
+            schema = gpd.io.file.infer_schema(geojson)
+            for col, dtype in schema["properties"].items():
+                if dtype in ("int", "int64"):
+                    schema["properties"][col] = "int32"
+            ogrgmt_kwargs["schema"] = schema
```

Comment:
If we're going to need to read the schema anyway, should we just use the pure `fiona` and no `geopandas` solution I mentioned at https://github.com/GenericMappingTools/pygmt/pull/1000#discussion_r606965331? Then we can remove the try-except block and the geopandas import?

Reply:
Sounds a good idea, although I have no experience with `__geo_interface__`.

Reply:
> If we're going to need to read the schema anyway, should we just use the pure `fiona` and no `geopandas` solution I mentioned at [#1000 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1000#discussion_r606965331)? Then we can remove the try-except block and the geopandas import?

I think we can't. Currently, we use the `gpd.io.file.infer_schema` function to infer the data schema, so we still have to import the geopandas package. More importantly, the `gpd.io.file.infer_schema` function only works for GeoPandas object.

Reply:
Yeah, let's just stick to using `gpd.io.file.infer_schema`, I think we'll have to because it contains the patch made at https://github.com/geopandas/geopandas/pull/2950 is in geopandas only and not fiona.